dashboard/test/controllers/api/v1/users_controller_test.rb

```diff
@@ -55,4 +55,24 @@ class Api::V1::UsersControllerTest < ActionController::TestCase
     @user.reload
     assert @user.ui_tip_dismissed_homepage_header
   end
+
+  test 'a post request to postpone_census_banner updates next_census_display' do
+    sign_in(@user)
+    @user.next_census_display = nil
+    assert @user.next_census_display.nil?
+    post :postpone_census_banner, params: {user_id: 'me'}
```

Comment:
1. No need to assert the thing you just set. I'd remove line 62
 2. Setting an attribute on the local `@user` object does not commit it to the DB, and will not make it available to the controller. Line 61 should be something like `@user.update!(next_census_display: nil)`, or you can create a new user object for each test case (in which case that field will already be nil).

Reply:
Same below

Reply:
I had modeled this after [an existing test](https://github.com/code-dot-org/code-dot-org/blob/d14c06590bd0642663a93ef4a972578ee279661f/dashboard/test/controllers/api/v1/users_controller_test.rb#L51). Look like that wasn't a great example. I will update based on your suggestions.

Reply:
Ick. That is a bad example. Adding a `reload` to this line https://github.com/code-dot-org/code-dot-org/blob/d14c06590bd0642663a93ef4a972578ee279661f/dashboard/test/controllers/api/v1/users_controller_test.rb#L51 will cause it to fail :(